locations/spiders/inpost_it.py

```diff
@@ -0,0 +1,110 @@
+import re
+
+from locations.categories import Categories, apply_category
+from locations.hours import CLOSED_IT, DAYS_IT, NAMED_DAY_RANGES_IT, NAMED_TIMES_IT, OpeningHours
+from locations.json_blob_spider import JSONBlobSpider
+
+
+class InpostITSpider(JSONBlobSpider):
+    name = "inpost_it"
+    allowed_domains = ["inpost.it"]
+    start_urls = []  # ["https://inpost.it/sites/default/files/points.json"]
+    locations_key = "items"
+    requires_proxy = True
+
+    operator = {"operator": "InPost", "operator_wikidata": "Q3182097"}
+    attributes = {"parcel_mail_in": "yes", "parcel_pickup": "yes"}
+    brand_locker = {"brand": "InPost", "brand_wikidata": "Q3182097", "nsi_id": "inpost-6b37ec"}
+    brand_partner = {"post_office:brand": "InPost", "post_office:brand:wikidata": "Q3182097"}
+
+    def start_requests(self):
+        for domain in self.allowed_domains:
+            self.start_urls.append(f"https://{domain}/sites/default/files/points.json")
+        yield from super().start_requests()
+
+    def pre_process_data(self, v):
+        # this mapping comes from "load" js function in inpost webpage
+        location = {
+            "position": {"lat": v["l"]["a"], "lng": v["l"]["o"]},
+            "street": v["e"],
+            "housenumber": v["b"],
+            "province": v["r"],
+            "post_code": v["o"],
+            "city": v["c"],
+            "type": v["t"],
+            "name": v["d"],  # originally, location_description
+            "hours": v["h"],
+            "payment": v["p"],
+            "ref": v["n"],  # originally, name
+            "status": v["s"],
+            "apm_doubled": v["m"],
+        }
+        v.clear()
+        v.update(location)
+        v["active"] = int(v["status"]) == 1
+        v["category"] = Categories.PARCEL_LOCKER if int(v["type"]) == 1 else Categories.POST_PARTNER
+
+    def post_process_item(self, item, response, location):
+        if not location["active"]:
+            return None
+
+        if hours := location["hours"]:
+            if hours != "24/7":
+                item["opening_hours"] = OpeningHours()
+                self.parse_hours(item["opening_hours"], hours)
```

Comment:
You probably want to collect 24/7? If not can you add a comment or someone else might.

Also I wouldn’t trust the output of `self.parse_hours` enough to edit OSM with it.

Reply:
Thanks (also for the commits), the losing of 24/7 was a refuse of the parsing in the DictParser.
I'm not sure I get the comment about `self.parse_hours` - it's up to the subclass to adapt it, like in the pl one where it is overwritten to use the PL days list, or am I not seeing something?

Reply:
I wouldn't trust the output of `add_ranges_from_string` enough to put into OSM.

Reply:
Ah in that sense, well usually I'd agree, but inpost formats them very well so they get converted nicely